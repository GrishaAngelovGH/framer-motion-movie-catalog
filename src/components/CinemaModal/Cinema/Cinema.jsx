const Cinema = ({ id, name, address, onClick }) => {
  const handleClick = () => {
    onClick(id)
  }

  return (
    <div onClick={handleClick} className="text-sm md:text-xl flex items-center p-2 border-b-2 hover:cursor-pointer hover:bg-gray-200">
      <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" strokeWidth={1.5} stroke="currentColor" className="size-10">
        <path strokeLinecap="round" strokeLinejoin="round" d="M15 10.5a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z" />
        <path strokeLinecap="round" strokeLinejoin="round" d="M19.5 10.5c0 7.142-7.5 11.25-7.5 11.25S4.5 17.642 4.5 10.5a7.5 7.5 0 1 1 15 0Z" />
      </svg>

      <div className="ms-2">
        <div>{name}</div>
        <div>{address}</div>
      </div>
    </div>
  )
}

export default Cinema