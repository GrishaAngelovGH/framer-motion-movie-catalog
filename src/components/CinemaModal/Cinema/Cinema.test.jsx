import { render } from "@testing-library/react"

import Cinema from "./Cinema"

test("should render Cinema component", () => {
  const view = render(<Cinema id="123" name="Cinema Name" address="Cinema Address" />)

  expect(view).toMatchSnapshot()
})