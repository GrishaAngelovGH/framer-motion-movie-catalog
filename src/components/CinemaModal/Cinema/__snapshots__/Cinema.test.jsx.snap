// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render Cinema component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-sm md:text-xl flex items-center p-2 border-b-2 hover:cursor-pointer hover:bg-gray-200"
        data-testid="cinema"
      >
        <svg
          class="size-10"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 10.5a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
          <path
            d="M19.5 10.5c0 7.142-7.5 11.25-7.5 11.25S4.5 17.642 4.5 10.5a7.5 7.5 0 1 1 15 0Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <div
          class="ms-2"
        >
          <div>
            Cinema Name
          </div>
          <div>
            Cinema Address
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-sm md:text-xl flex items-center p-2 border-b-2 hover:cursor-pointer hover:bg-gray-200"
      data-testid="cinema"
    >
      <svg
        class="size-10"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 10.5a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
        <path
          d="M19.5 10.5c0 7.142-7.5 11.25-7.5 11.25S4.5 17.642 4.5 10.5a7.5 7.5 0 1 1 15 0Z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <div
        class="ms-2"
      >
        <div>
          Cinema Name
        </div>
        <div>
          Cinema Address
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
